From 5dc2b9a5c09d95ba9a17ac00056213082feb8d90 Mon Sep 17 00:00:00 2001
From: Raymond Wooninck <tittiatcoke@gmail.com>
Date: Tue, 25 Feb 2014 19:59:36 +0100
Subject: [PATCH] Perform the systemd version evaluation through a regular
 expression to be more flexible for changes in the version string that systemd
 returns

REVIEW: 115271
---
 .../daemon/backends/upower/powerdevilupowerbackend.cpp       | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/powerdevil/daemon/backends/upower/powerdevilupowerbackend.cpp b/powerdevil/daemon/backends/upower/powerdevilupowerbackend.cpp
index eae5436..94b20f1 100644
--- a/powerdevil/daemon/backends/upower/powerdevilupowerbackend.cpp
+++ b/powerdevil/daemon/backends/upower/powerdevilupowerbackend.cpp
@@ -40,13 +40,17 @@
 
 bool checkSystemdVersion(uint requiredVersion)
 {
-    bool ok;
 
     QDBusInterface systemdIface("org.freedesktop.systemd1", "/org/freedesktop/systemd1", "org.freedesktop.systemd1.Manager",
                                 QDBusConnection::systemBus(), 0);
-    const uint version = systemdIface.property("Version").toString().section(' ', 1).toUInt(&ok);
-    if (ok) {
-       return (version >= requiredVersion);
+
+    const QString reply = systemdIface.property("Version").toString();
+
+    QRegExp expsd("(systemd )?([0-9]+)");
+
+    if (expsd.exactMatch(reply)) {
+        const uint version = expsd.cap(2).toUInt();
+        return (version >= requiredVersion);
     }
 
     // Since version 1.11 Upstart user sessions implement the exact same API as logind
-- 
1.9.0

